### PR TITLE
Upsss... a big pull-request

### PR DESCRIPTION
**Pull request by gmarcos87**
After we stopped creating issues I continued working on this branch. That made the pull-request much longer than desired and with several things at once.
Basically:
- Solve some bugs found in the way
- Add routes for configuration
- The db is loaded asynchronously
- The configuration is saved in .config.json and also via shared-state

### CODE REVIEW

File: federated-server/shcedule/nodes.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import pull from 'pull-stream';
```
Comment (germanferrero) [inline federated-server/shcedule/nodes.js]: s/shcedule/schedule/g  in directory name

File: federated-server/utils/faltToUnique.js [hunk 2]
```diff
@@ -1 +1 @@
-export const flatToUnique = (prev,act) => [...prev, ...act.filter(node => prev.indexOf(node ) === -1)]
+export const flatToUnique = (prev=[],act) => [...prev, ...act.filter(node => prev.indexOf(node ) === -1)]
```
Comment (germanferrero) [inline federated-server/utils/faltToUnique.js]: s/faltToUnique/flatToUnique/g in filename

File: federated-server/db.js [hunk 3]
```diff
@@ -1,42 +1,58 @@
...
-    console.log('Using room invite', process.env.ROOM_INVITE)
-    let address = utils.inviteToAddress(process.env.ROOM_INVITE)
-    sbot.conn.connect(address, {type: 'room'}, (err)=>console.log(err? 'ERROR: '+err: 'Conexión con ssb-room correcta'))
+function startPeerConnections() {
+  setTimeout(()=>{
+    sbot.conn.start()  
+    if( process.env.ROOM_INVITE) {
+      console.log('Using room invite', process.env.ROOM_INVITE)
+      let address = utils.inviteToAddress(process.env.ROOM_INVITE)
+      sbot.conn.connect(address, {type: 'room'}, (err)=>console.log(err? 'ERROR: '+err: 'Conexión con ssb-room correcta'))
```
Comment (germanferrero) [inline federated-server/db.js]: Let's keep logs in english

File: federated-server/shared-state.js [hunk 4]
```diff
@@ -32,19 +30,19 @@ export async function getActualNodes(){
...
+    let servers = []
+    myState
+        .show('soport-servers')
+        .forEach(({data}) => { servers = [...servers, data] })
+        return servers;
 }
 
+export async function sendSharedConfig(config) {
+    console.log(config)
+    myState.insert(getIps(), {config}, 100, 'soport-servers')
```
[the remark this answers is not in the document]
Comment (gmarcos87) [reply federated-server/shared-state.js]: done

File: federated-server/utils/faltToUnique.js (hunk 2 above)
Comment (gmarcos87) [reply federated-server/utils/faltToUnique.js]: done

File: federated-server/shcedule/nodes.js (hunk 1 above)
Comment (gmarcos87) [reply federated-server/shcedule/nodes.js]: done

File: federated-server/db.js (hunk 3 above)
Comment (gmarcos87) [reply federated-server/db.js]: done

**Pull request merged**